bigquery/src/main/java/org/apache/iceberg/gcp/bigquery/BigQueryMetastoreClientImpl.java

```diff
@@ -120,13 +120,12 @@ public RetryResult beforeEval(Exception exception) {
           .build();
 
   /** Constructs a client of the Google BigQuery service. */
-  public BigQueryMetastoreClientImpl(BigQueryOptions options)
+  public BigQueryMetastoreClientImpl(GoogleCredentials credentials, BigQueryOptions options)
       throws IOException, GeneralSecurityException {
     // Initialize client that will be used to send requests. This client only needs to be created
     // once, and can be reused for multiple requests
     HttpCredentialsAdapter httpCredentialsAdapter =
-        new HttpCredentialsAdapter(
-            GoogleCredentials.getApplicationDefault().createScoped(BigqueryScopes.all()));
+        new HttpCredentialsAdapter(credentials.createScoped(BigqueryScopes.all()));
```

Comment:
@ebyhr Do yout hink we should also pass credentails for FileIO ? With this change FileIO and Bigquery may end up  different credentials fyi